2371/2/battle_entity.rb

```diff
@@ -0,0 +1,22 @@
+# The BattleEntity responsible for battle info
+class BattleEntity
+  def initialize(file_name)
+    @curse_words = []
+    @words_percent = 0
+    manage_battle_text(File.read("./texts/#{file_name}"))
```

Comment:
No logic in initializers!
Think about the messages which you will pass to the `Battle` object.

My suggestion

```ruby
# battle.rb
class Battle
  attr_reader :battle_file_path, ... #etc
  
  def intialize(battle_file_path)
     # ...
  end

  def words_count; end
  def rounds_count; end
  def bad_words_count; end
  def battler_name; end
  # ...
end
```